client/my-sites/email/email-providers-stacked-comparison/index.tsx

```diff
@@ -86,6 +65,28 @@ const EmailProvidersStackedComparison = ( {
 		};
 	} );
 
+	const canPurchaseGoogleWorkspace = useSelector( canUserPurchaseGSuite );
+
+	const currentRoute = useSelector( getCurrentRoute );
+
+	const selectedSite = useSelector( getSelectedSite );
+
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSite?.ID ) );
+	const domain = getSelectedDomain( {
+		domains,
+		selectedDomainName: selectedDomainName,
+	} );
+	const domainsWithForwards = useSelector( ( state ) => getDomainsWithForwards( state, domains ) );
+
+	const resolvedDomainName = domain ? domain.name : selectedDomainName;
+
+	if ( ! resolvedDomainName ) {
+		return <></>;
+	}
```

Comment:
I think this will prevent us to show errors.

Reply:
Could you clarify what you mean? What happens if you don't add this condition? Moreover, as far as I can see we should always have at least `selectedDomainName` so I'm not sure when this `resolvedDomainName` would be falsy.

Reply:
You are right, I was recalling on an issue we had a few months ago were `domain.name` couldn't be resolved and sometimes the old comparison failed to load/render. But you are right, this time resolvedDomainName won't be falsy at any case as we ask for a mandatory prop called `selectedDomainName`

Reply:
@stephanethomas Oh just remembered, I added that guard actually not because of the resolvedDomainName, but just in case domain has not been loaded yet, and it can crash the screen.

Reply:
We can make the screen to fail (and see an error page) if we are in that page and we clear site data in Chrome debugger.